scripts/spec.py

```diff
@@ -56,17 +56,26 @@
 # these are specifications of objects with fields that are marked to have a union type of IntOrString. However, K8s
 # only accepts one or the other, unfortunately. Here, we remap those fields from their respective `$ref`s, which
 # contain a specific type, to another type. The mapping is from the `$ref` to a tuple of the existing type and the
-# new type. The dictionary model is:
+# new type. While the first piece of the mapping tuple must be defined the second one is optional - if the second
+# part of the tuple is None then the field will be removed. The dictionary model is:
 # { object name: { field name: ( ( existing field, existing value ) , ( new field, new value ) ) } }
-INT_OR_STRING_FIELD_REMAPPING: Dict[str, Dict[str, Tuple[Tuple[str, str], Tuple[str, str]]]] = {
+FIELD_REMAPPINGS: Dict[
+    str, Dict[str, Tuple[Tuple[str, Union[str, List[str]]], Optional[Tuple[str, Union[str, List[str]]]]]]
+] = {
     "io.k8s.api.core.v1.HTTPGetAction": {
         "port": (
             ("$ref", "#/definitions/io.k8s.apimachinery.pkg.util.intstr.IntOrString"),
             ("type", "integer"),
         ),
     },
+    "io.k8s.api.core.v1.Container": {
+        "imagePullPolicy": (
+            ("enum", None),
+            None,
```

Comment:
What's this `None` doing here? 🤔

Reply:
First part of the tuple says "original field, original field value" second one says "new field, new field value". New field value of None means delete the original field. So, this is removing the enum from Container, allowing codegen to put a string in place of the Container image pull policy enum, and stop generating the ImagePullPolicy enum altogether.

Later in the script you see we have to add a specification for IPP manually for otherwise it's not fully backwards compatible, even if it's not used by Container!

Reply:
Ah of course, I was misreading the outer tuple. Makes sense (almost!), though I would suggest a refactor to helper classes if you have time/want to in a separate PR